commit a5ccf1fc09eb89ea5cce250198f24857d63ec19f
Author: jouyouyun <jouyouwen717@gmail.com>
Date:   Tue Feb 7 14:40:15 2017 +0800

    nm_generator: Add quote on string constants
    
    Some constants, such as 'pkcs11:' can cause YAML unmarshal failed, so add quote on all string constants.
    
    Change-Id: I9a900ac1101591e3ba439a02a91cc6b44ce3a46f

diff --git a/network/nm_generator/gen_nm_consts.py b/network/nm_generator/gen_nm_consts.py
index 98ba6fad..88b0e3d0 100755
--- a/network/nm_generator/gen_nm_consts.py
+++ b/network/nm_generator/gen_nm_consts.py
@@ -201,6 +201,6 @@ for const in girxml.findall('./gi:namespace/gi:constant', ns_map):
     cvalue = const.attrib['value']
     if cname not in constants and const.find('./gi:type[@c:type="gchar*"]', ns_map) is not None:
         constants[cname]=cvalue
-        outfile.write("    - Name: %s\n      Value: %s\n" % (cname, cvalue))
+        outfile.write("    - Name: %s\n      Value: \"%s\"\n" % (cname, cvalue))
 
 outfile.close()
diff --git a/network/nm_setting_beans_extend.go b/network/nm_setting_beans_extend.go
index addc3b13..ea5637b4 100644
--- a/network/nm_setting_beans_extend.go
+++ b/network/nm_setting_beans_extend.go
@@ -91,6 +91,16 @@ func checkSettingIPTunnelValues(data connectionData) (err sectionErrors) {
 	return
 }
 
+func getSettingMacsecAvailableKeys(data connectionData) (keys []string) {
+	return
+}
+func getSettingMacsecAvailableValues(data connectionData, key string) (values []kvalue) {
+	return
+}
+func checkSettingMacsecValues(data connectionData) (err sectionErrors) {
+	return
+}
+
 func getSettingMacvlanAvailableKeys(data connectionData) (keys []string) {
 	return
 }
@@ -111,6 +121,16 @@ func checkSettingOlpcMeshValues(data connectionData) (err sectionErrors) {
 	return
 }
 
+func getSettingProxyAvailableKeys(data connectionData) (keys []string) {
+	return
+}
+func getSettingProxyAvailableValues(data connectionData, key string) (values []kvalue) {
+	return
+}
+func checkSettingProxyValues(data connectionData) (err sectionErrors) {
+	return
+}
+
 func getSettingTeamAvailableKeys(data connectionData) (keys []string) {
 	return
 }
